osu.Game.Rulesets.Mania/Mods/ManiaModFadeIn.cs

```diff
@@ -19,5 +30,103 @@ public class ManiaModFadeIn : Mod
+
+                hocParent.Remove(hoc);
+                hocParent.Add(new BufferedContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        hoc,
+                        new LaneCover(0.5f, false)
+                        {
+                            RelativeSizeAxes = Axes.Both
+                        }
+                    }
+                });
+            }
+        }
+
+        private class LaneCover : CompositeDrawable
+        {
+            private readonly Box gradient;
+            private readonly Box filled;
+
+            public LaneCover(float initialCoverage, bool reversed)
+            {
+                Blending = new BlendingParameters
+                {
+                    RGBEquation = BlendingEquation.Add,
+                    Source = BlendingType.Zero,
+                    Destination = BlendingType.One,
+                    AlphaEquation = BlendingEquation.Add,
+                    SourceAlpha = BlendingType.Zero,
+                    DestinationAlpha = BlendingType.OneMinusSrcAlpha
+                };
+                InternalChildren = new Drawable[]
+                {
+                    gradient = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        RelativePositionAxes = Axes.Both,
+                        Height = 0.25f
+                    },
+                    filled = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both
+                    }
+                };
+                Coverage = initialCoverage;
+                Reversed = reversed;
+            }
+
+            private float coverage;
+
+            public float Coverage
+            {
+                set
+                {
+                    filled.Height = value;
+                    gradient.Y = reversed ? 1 - value - gradient.Height : value;
+                    coverage = value;
+                }
+            }
+
+            private bool reversed;
+
+            public bool Reversed
+            {
+                set
+                {
+                    filled.Anchor = value ? Anchor.BottomLeft : Anchor.TopLeft;
+                    filled.Origin = value ? Anchor.BottomLeft : Anchor.TopLeft;
+                    gradient.Colour = ColourInfo.GradientVertical(
+                        Color4.White.Opacity(value ? 0f : 1f),
+                        Color4.White.Opacity(value ? 1f : 0f)
+                    );
+
+                    reversed = value;
+                    Coverage = coverage; //re-apply coverage to update visuals
+                }
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(ManiaRulesetConfigManager configManager)
+            {
+                var scrollDirection = configManager.GetBindable<ManiaScrollingDirection>(ManiaRulesetSetting.ScrollDirection);
```

Comment:
This bindable needs to be stored, otherwise the binding will be lost. Do something like:

```csharp
private readonly IBindable<ManiaScrollingDirection> scrollDirection = new Bindable<ManiaScrollingDirection>();

private void load()
{
    scrollDirection.BindTo(otherBindable);
}
```

I think `otherBindable` in this case should come from `IScrollingInfo`. See how `HitObjectArea` does things for reference.